### ISSUE TITLE
TFVC jobs in parallel on the hosted agent may try to delete the same workspace at the exact same time.

### ISSUE THREAD

**Issue by jessehouwing** [microsoft/azure-pipelines-agent#3588]
## Agent Version and Platform
Version of your agent? 2.194.0'

OS of the machine running the agent? Windows

## Azure DevOps Type and Version
dev.azure.com 

If dev.azure.com, what is your organization name? can't share

## What's not working?
During checkout the agent queries workspaces assigned to the same machine name and subsequently tries to delete these. Due to the fact that VMs in the hosted pool have the same machine name in certain cases, when running build in parallel 2 builds may hit this exact codepath at the same time.

When this happens, one build will succeed and the other will fail to delete the workspace, because it no longer exists.

```
##[error]Exit code 100 returned from process: file name 'tf', arguments 'vc workspace /delete ws_1_326;c75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/org/ /loginType:OAuth /login:.,*** /noprompt'.
##[debug]Processed: ##vso[task.logissue type=error;]Exit code 100 returned from process: file name 'tf', arguments 'vc workspace /delete ws_1_326%3Bc75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/org/ /loginType:OAuth /login:.,*** /noprompt'.
##[debug]Processed: ##vso[task.complete result=Failed;]
##[debug]   at Microsoft.VisualStudio.Services.Agent.Util.ProcessInvoker.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, InputQueue`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
   at Agent.Plugins.Repository.TfsVCCliManager.RunCommandAsync(FormatFlags formatFlags, Boolean quiet, Boolean failOnNonZeroExitCode, String[] args)
   at Agent.Plugins.Repository.TfsVCCliManager.RunCommandAsync(FormatFlags formatFlags, Boolean quiet, Int32 retriesOnFailure, String[] args)
   at Agent.Plugins.Repository.TFCliManager.WorkspaceDeleteAsync(ITfsVCWorkspace workspace)
   at Agent.Plugins.Repository.TfsVCSourceProvider.GetSourceAsync(AgentTaskPluginExecutionContext executionContext, RepositoryResource repository, CancellationToken cancellationToken)
   at Agent.Plugins.Repository.CheckoutTask.RunAsync(AgentTaskPluginExecutionContext executionContext, CancellationToken token)
   at Agent.PluginHost.Program.Main(String[] args)
```

Ideally the agent would retry the querying of workspaces and the following delete action.

## Agent and Worker's Diagnostic Logs
```
2021-10-29T22:56:46.4254669Z ##[debug]Evaluating condition for step: 'Checkout pppppppto s'
2021-10-29T22:56:46.4257411Z ##[debug]Evaluating: SucceededNode()
2021-10-29T22:56:46.4257954Z ##[debug]Evaluating SucceededNode:
2021-10-29T22:56:46.4259016Z ##[debug]=> True
2021-10-29T22:56:46.4259540Z ##[debug]Result: True
2021-10-29T22:56:46.4260094Z ##[section]Starting: Checkout ppppppp@ to s
2021-10-29T22:56:46.4423044Z ==============================================================================
2021-10-29T22:56:46.4423326Z Task         : Get sources
2021-10-29T22:56:46.4423585Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2021-10-29T22:56:46.4423880Z Version      : 1.0.0
2021-10-29T22:56:46.4424069Z Author       : Microsoft
2021-10-29T22:56:46.4424336Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2021-10-29T22:56:46.4424646Z ==============================================================================
2021-10-29T22:56:46.8825848Z ##[debug]Processed: ##vso[plugininternal.updaterepositorypath alias=__designer_repo;]D:\a\1\s
2021-10-29T22:56:46.8827738Z ##[debug]Repository requires to be placed at 'D:\a\1\s', current location is 'D:\a\1\s'
2021-10-29T22:56:46.9221350Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework', value name 'InstallRoot': 'C:\Windows\Microsoft.NET\Framework64\'
2021-10-29T22:56:46.9251723Z ##[debug]Key name 'SOFTWARE\Microsoft\NET Framework Setup\NDP' contains sub keys:
2021-10-29T22:56:46.9252696Z ##[debug] 'CDF'
2021-10-29T22:56:46.9253412Z ##[debug] 'v2.0.50727'
2021-10-29T22:56:46.9254245Z ##[debug] 'v3.0'
2021-10-29T22:56:46.9255032Z ##[debug] 'v3.5'
2021-10-29T22:56:46.9255706Z ##[debug] 'v4'
2021-10-29T22:56:46.9256623Z ##[debug] 'v4.0'
2021-10-29T22:56:46.9257625Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v2.0.50727', value name 'Version': '2.0.50727.4927'
2021-10-29T22:56:46.9265594Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v2.0.50727', value name 'Install': '1'
2021-10-29T22:56:46.9266959Z ##[debug]Testing directory: 'C:\Windows\Microsoft.NET\Framework64\v2.0.50727'
2021-10-29T22:56:46.9283203Z ##[debug]Found version: 2.0.50727
2021-10-29T22:56:46.9284296Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.0', value name 'Version': '3.0.30729.4926'
2021-10-29T22:56:46.9285436Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.0', value name 'Install': '1'
2021-10-29T22:56:46.9286598Z ##[debug]Testing directory: 'C:\Windows\Microsoft.NET\Framework64\v3.0'
2021-10-29T22:56:46.9287449Z ##[debug]Found version: 3.0
2021-10-29T22:56:46.9288495Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.5', value name 'Version': '3.5.30729.4926'
2021-10-29T22:56:46.9289706Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v3.5', value name 'Install': '1'
2021-10-29T22:56:46.9290744Z ##[debug]Testing directory: 'C:\Windows\Microsoft.NET\Framework64\v3.5'
2021-10-29T22:56:46.9291536Z ##[debug]Found version: 3.5
2021-10-29T22:56:46.9292521Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4', value name 'Version' is null.
2021-10-29T22:56:46.9293550Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4', value name '' is null.
2021-10-29T22:56:46.9294567Z ##[debug]Key name 'SOFTWARE\Microsoft\NET Framework Setup\NDP\v4' contains sub keys:
2021-10-29T22:56:46.9295407Z ##[debug] 'Client'
2021-10-29T22:56:46.9296107Z ##[debug] 'Full'
2021-10-29T22:56:46.9297610Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Client', value name 'Version': '4.8.03761'
2021-10-29T22:56:46.9298796Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Client', value name 'Install': '1'
2021-10-29T22:56:46.9299979Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Client', value name 'InstallPath': 'C:\Windows\Microsoft.NET\Framework64\v4.0.30319\'
2021-10-29T22:56:46.9303014Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Client', value name 'Release': '528049'
2021-10-29T22:56:46.9303905Z ##[debug]Type is System.Int32
2021-10-29T22:56:46.9304766Z ##[debug]Interpreted version: 4.7.0
2021-10-29T22:56:46.9305801Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full', value name 'Version': '4.8.03761'
2021-10-29T22:56:46.9307141Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full', value name 'Install': '1'
2021-10-29T22:56:46.9308315Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full', value name 'InstallPath': 'C:\Windows\Microsoft.NET\Framework64\v4.0.30319\'
2021-10-29T22:56:46.9309530Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full', value name 'Release': '528049'
2021-10-29T22:56:46.9310490Z ##[debug]Type is System.Int32
2021-10-29T22:56:46.9311400Z ##[debug]Interpreted version: 4.7.0
2021-10-29T22:56:46.9312295Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4.0', value name 'Version' is null.
2021-10-29T22:56:46.9313384Z ##[debug]Key name 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4.0', value name '': 'deprecated'
2021-10-29T22:56:46.9314290Z ##[debug]Found 5 versions:
2021-10-29T22:56:46.9315087Z ##[debug] 2.0.50727
2021-10-29T22:56:46.9315780Z ##[debug] 3.0
2021-10-29T22:56:46.9316663Z ##[debug] 3.5
2021-10-29T22:56:46.9317448Z ##[debug] 4.7.0
2021-10-29T22:56:46.9318140Z ##[debug] 4.7.0
2021-10-29T22:56:46.9319013Z ##[debug]Testing for min NET Framework version: '4.6'
2021-10-29T22:56:46.9439217Z Prepending Path environment variable with directory containing 'tf.exe'.
2021-10-29T22:56:46.9451301Z ##[debug]Processed: ##vso[task.prependpath]C:\agents\2.194.0\externals\tf
2021-10-29T22:56:46.9548269Z ##[debug]PATH: 'C:\agents\2.194.0\externals\tf;C:\Users\VssAdministrator\.dotnet\tools;C:\Program Files\MongoDB\Server\5.0\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS\;C:\tools\zstd;C:\Program Files\Mercurial\;C:\hostedtoolcache\windows\stack\2.7.3\x64;C:\tools\ghc-9.0.1\bin;C:\Program Files\dotnet;C:\mysql\bin;C:\Program Files\R\R-4.1.1\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Rust\.cargo\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\hostedtoolcache\windows\go\1.15.15\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.9\x64\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.302-8\x64\bin;C:\npm\prefix;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\Chocolatey\bin;C:\Program Files\Docker;C:\Program Files\PowerShell\7\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\;C:\Program Files\nodejs\;C:\Program Files\LLVM\bin;C:\Program Files\OpenSSL\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\TortoiseSVN\bin;C:\Program Files\CMake\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.3\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;c:\tools\php;C:\Program Files (x86)\sbt\bin;C:\SeleniumWebDrivers\ChromeDriver\;C:\SeleniumWebDrivers\EdgeDriver\;C:\Program Files\Amazon\AWSCLIV2\;C:\Program Files\Amazon\SessionManagerPlugin\bin\;C:\Program Files\Amazon\AWSSAMCLI\bin\;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server\;C:\Users\VssAdministrator\AppData\Local\Microsoft\WindowsApps'
2021-10-29T22:56:46.9555288Z Setting environment variable TFVC_BUILDAGENT_POLICYPATH
2021-10-29T22:56:46.9564597Z ##[debug]Processed: ##vso[task.setvariable variable=TFVC_BUILDAGENT_POLICYPATH;issecret=False;]C:\agents\2.194.0\externals\tf\Microsoft.TeamFoundation.VersionControl.Controls.dll
2021-10-29T22:56:46.9565737Z Querying workspace information.
2021-10-29T22:56:46.9593158Z ##[debug]tf vc workspaces /format:xml /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt
2021-10-29T22:56:46.9708524Z ##[debug]Starting process:
2021-10-29T22:56:46.9709352Z ##[debug]  File name: 'tf'
2021-10-29T22:56:46.9712923Z ##[debug]  Arguments: 'vc workspaces /format:xml /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt'
2021-10-29T22:56:46.9713947Z ##[debug]  Working directory: 'D:\a\1\s'
2021-10-29T22:56:46.9715702Z ##[debug]  Require exit code zero: 'True'
2021-10-29T22:56:46.9716826Z ##[debug]  Encoding web name: windows-1252 ; code page: '1252'
2021-10-29T22:56:46.9717761Z ##[debug]  Force kill process on cancellation: 'False'
2021-10-29T22:56:46.9718574Z ##[debug]  Redirected STDIN: 'False'
2021-10-29T22:56:46.9719653Z ##[debug]  Persist current code page: 'False'
2021-10-29T22:56:46.9720468Z ##[debug]  Keep redirected STDIN open: 'False'
2021-10-29T22:56:46.9721595Z ##[debug]  High priority process: 'False'
2021-10-29T22:56:46.9988114Z ##[debug]OOM score adjustment is Linux-only.
2021-10-29T22:56:47.0084060Z ##[debug]Process started with process id 3512, waiting for process exit.
2021-10-29T22:56:49.1056241Z ##[debug]<Workspaces />
2021-10-29T22:56:49.1319374Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:49.1320748Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:49.1420249Z ##[debug]Exited process 3512 with exit code 0
2021-10-29T22:56:49.1422478Z ##[debug]Finished process 3512 with exit code 0, and elapsed time 00:00:02.1697651.
2021-10-29T22:56:49.2358451Z ##[debug]Processed: ##vso[task.setvariable variable=build.repository.tfvc.workspace;issecret=False;]ws_1_326
2021-10-29T22:56:49.2513141Z ##[debug]Sources directory does not exist or is empty.
2021-10-29T22:56:49.2554549Z ##[debug]tf vc workspaces ws_1_326 /computer:* /format:xml /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt
2021-10-29T22:56:49.2555939Z ##[debug]Starting process:
2021-10-29T22:56:49.2556740Z ##[debug]  File name: 'tf'
2021-10-29T22:56:49.2560191Z ##[debug]  Arguments: 'vc workspaces ws_1_326 /computer:* /format:xml /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt'
2021-10-29T22:56:49.2561107Z ##[debug]  Working directory: 'D:\a\1\s'
2021-10-29T22:56:49.2561881Z ##[debug]  Require exit code zero: 'True'
2021-10-29T22:56:49.2562566Z ##[debug]  Encoding web name: windows-1252 ; code page: '1252'
2021-10-29T22:56:49.2563405Z ##[debug]  Force kill process on cancellation: 'False'
2021-10-29T22:56:49.2564382Z ##[debug]  Redirected STDIN: 'False'
2021-10-29T22:56:49.2565221Z ##[debug]  Persist current code page: 'False'
2021-10-29T22:56:49.2565727Z ##[debug]  Keep redirected STDIN open: 'False'
2021-10-29T22:56:49.2566215Z ##[debug]  High priority process: 'False'
2021-10-29T22:56:49.2581495Z ##[debug]OOM score adjustment is Linux-only.
2021-10-29T22:56:49.2582787Z ##[debug]Process started with process id 4228, waiting for process exit.
2021-10-29T22:56:50.5054284Z ##[debug]<Workspaces>
2021-10-29T22:56:50.5063224Z ##[debug]  <Workspace computer="fv-az423-207" islocal="true" name="ws_1_326" ownerdisp="Project Collection Build Service (OrgS)" ownerid="ab21bc76-eb45-4c55-bc13-70ab5137d5ad:Build:03bfd66a-1ced-4a3f-b479-186dc2c894f0" ownertype="Microsoft.TeamFoundation.ServiceIdentity" owner="c75ebe8d-c40f-466b-8698-55694170789e" owneruniq="c75ebe8d-c40f-466b-8698-55694170789e">
2021-10-29T22:56:50.5065040Z ##[debug]    <Comment />
2021-10-29T22:56:50.5068683Z ##[debug]    <Folders>
2021-10-29T22:56:50.5069294Z ##[debug]      <WorkingFolder local="D:\a\1\s" item="$/xxxx" />
2021-10-29T22:56:50.5069840Z ##[debug]    </Folders>
2021-10-29T22:56:50.5070747Z ##[debug]    <LastAccessDate>2021-10-29T22:55:06.777+00:00</LastAccessDate>
2021-10-29T22:56:50.5071567Z ##[debug]    <OwnerAliases>
2021-10-29T22:56:50.5072258Z ##[debug]      <string>Build\03bfd66a-1ced-4a3f-b479-186dc2c894f0</string>
2021-10-29T22:56:50.5073355Z ##[debug]      <string>03bfd66a-1ced-4a3f-b479-186dc2c894f0</string>
2021-10-29T22:56:50.5074239Z ##[debug]      <string>Project Collection Build Service (OrgS)</string>
2021-10-29T22:56:50.5075036Z ##[debug]    </OwnerAliases>
2021-10-29T22:56:50.5075741Z ##[debug]  </Workspace>
2021-10-29T22:56:50.5076216Z ##[debug]</Workspaces>
2021-10-29T22:56:50.5212932Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:50.5217389Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:50.5325413Z ##[debug]Exited process 4228 with exit code 0
2021-10-29T22:56:50.5327742Z ##[debug]Finished process 4228 with exit code 0, and elapsed time 00:00:01.2765377.
2021-10-29T22:56:50.5456050Z ##[command]tf vc workspace /delete ws_1_326%3Bc75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt
2021-10-29T22:56:50.5458725Z ##[debug]Starting process:
2021-10-29T22:56:50.5460452Z ##[debug]  File name: 'tf'
2021-10-29T22:56:50.5464854Z ##[debug]  Arguments: 'vc workspace /delete ws_1_326;c75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt'
2021-10-29T22:56:50.5466011Z ##[debug]  Working directory: 'D:\a\1\s'
2021-10-29T22:56:50.5468106Z ##[debug]  Require exit code zero: 'True'
2021-10-29T22:56:50.5468967Z ##[debug]  Encoding web name: windows-1252 ; code page: '1252'
2021-10-29T22:56:50.5469849Z ##[debug]  Force kill process on cancellation: 'False'
2021-10-29T22:56:50.5470640Z ##[debug]  Redirected STDIN: 'False'
2021-10-29T22:56:50.5471958Z ##[debug]  Persist current code page: 'False'
2021-10-29T22:56:50.5472781Z ##[debug]  Keep redirected STDIN open: 'False'
2021-10-29T22:56:50.5473344Z ##[debug]  High priority process: 'False'
2021-10-29T22:56:50.5479654Z ##[debug]OOM score adjustment is Linux-only.
2021-10-29T22:56:50.5483139Z ##[debug]Process started with process id 472, waiting for process exit.
2021-10-29T22:56:51.2881428Z TF14061: The workspace ws_1_326;Project Collection Build Service (OrgS) does not exist.
2021-10-29T22:56:51.3060812Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:51.3063319Z ##[debug]STDOUT/STDERR stream read finished.
2021-10-29T22:56:51.3162414Z ##[debug]Exited process 472 with exit code 100
2021-10-29T22:56:51.3163571Z ##[debug]Finished process 472 with exit code 100, and elapsed time 00:00:00.7704586.
2021-10-29T22:56:51.3300807Z ##[error]Exit code 100 returned from process: file name 'tf', arguments 'vc workspace /delete ws_1_326;c75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt'.
2021-10-29T22:56:51.3313972Z ##[debug]Processed: ##vso[task.logissue type=error;]Exit code 100 returned from process: file name 'tf', arguments 'vc workspace /delete ws_1_326%3Bc75ebe8d-c40f-466b-8698-55694170789e /collection:https://dev.azure.com/OrgS/ /loginType:OAuth /login:.,*** /noprompt'.
2021-10-29T22:56:51.3317017Z ##[debug]Processed: ##vso[task.complete result=Failed;]
2021-10-29T22:56:51.3841380Z ##[debug]   at Microsoft.VisualStudio.Services.Agent.Util.ProcessInvoker.ExecuteAsync(String workingDirectory, String fileName, String arguments, IDictionary`2 environment, Boolean requireExitCodeZero, Encoding outputEncoding, Boolean killProcessOnCancel, InputQueue`1 redirectStandardIn, Boolean inheritConsoleHandler, Boolean keepStandardInOpen, Boolean highPriorityProcess, CancellationToken cancellationToken)
   at Agent.Plugins.Repository.TfsVCCliManager.RunCommandAsync(FormatFlags formatFlags, Boolean quiet, Boolean failOnNonZeroExitCode, String[] args)
   at Agent.Plugins.Repository.TfsVCCliManager.RunCommandAsync(FormatFlags formatFlags, Boolean quiet, Int32 retriesOnFailure, String[] args)
   at Agent.Plugins.Repository.TFCliManager.WorkspaceDeleteAsync(ITfsVCWorkspace workspace)
   at Agent.Plugins.Repository.TfsVCSourceProvider.GetSourceAsync(AgentTaskPluginExecutionContext executionContext, RepositoryResource repository, CancellationToken cancellationToken)
   at Agent.Plugins.Repository.CheckoutTask.RunAsync(AgentTaskPluginExecutionContext executionContext, CancellationToken token)
   at Agent.PluginHost.Program.Main(String[] args)
2021-10-29T22:56:51.4129523Z ##[section]Finishing: Checkout pppppppto s
```

```

**Comment by jessehouwing** [comment]
@anatolybolshakov

**Comment by jessehouwing** [comment]
@vijayma Any chance this one can be merged? It's causing build failures when running builds in parallel on the hosted agent.

**Comment by anatolybolshakov** [comment]
Reopened to keep it till changes are being rolled out.

**Comment by anatolybolshakov** [comment]
Hi @jessehouwing just as a quick update - we've reverted changes recently due to issue appeared https://github.com/microsoft/azure-pipelines-agent/issues/3638 and prepared a fix - we are going to continue rolling out of these changes with the fix applied, please note that there could be some delays in roll out due to New Year holidays.

**Comment by jessehouwing** [comment]
ARGH! :).

My extension is still keeping most pipelines working...

**Comment by anatolybolshakov** [comment]
Hi @jessehouwing we've rolled out the fix for the issue - please let us know if you still see any issues.
Closing this at the moment since fix has been already rolled out.

**Comment by jessehouwing** [comment]
This is *not* fixed with 2.196.2

**Comment by jessehouwing** [comment]
![IMG-20220118-WA0002~2](https://user-images.githubusercontent.com/4173387/149917467-ea3f5c15-2307-4284-b167-0c400b54b48e.jpg)
